src/main/java/my/test/shop/driver/Driver.java

```diff
@@ -0,0 +1,87 @@
+package my.test.shop.driver;
+
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.remote.AbstractDriverOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Driver {
```

Comment:
1. Your implementation has eager initialization of firefox and chrome... This is overhead. I recommend to have lazy initialization of needed browser only.
2. Your code will be impossible to run using local browser